hw/application_fpga/core/tk1/README.md

```diff
@@ -111,9 +111,21 @@ secret for any secrets it needs to perform its intended use case.
 ```
 
 These registers provide read access to the 64-bit unique device
-identity. The UDI is stored as ROM within the FPGA configuration. The
-registers can't be written to.
+identity. The registers can't be written to.
 
+The 2 UDI words are stored using 32 named LUT4s identified by
+"udi\_rom\_idx".
+
+We use bit indexing from all of the 32 LUTs to store each bit of the
+word, i.e., the first UDI word consists of bit 0 from each 32 LUTs,
+the second word consists of bit 1 from each of the 32 LUTs, and so
+on... for a maximum storage of 16 words, but we use only 2.
+
+The UDI does not occupy the entire LUT4 instance and to preserve
+hardware resources and not having to care about all the address bits
+the UDI should be repeated over the unused portion of the LUT4
+instance. This eliminates the need to drive the three MSB pins into
+the LUT4 while still achieving the correct output.
 
```

Comment:
I don't agree on the preserve HW resources phrasing. In fact we are wasting 32 * 14 bits by using this setup. We could have used four LUT4 and then muxes and a couple of registers. But that would affect performance and latency.

I would rephrase to something that:
- States that 32 explictly instantiated are used, one for each bit in the data word the CPU reads.
- Each LUT4 can store 16-bits. But since the UDI is 64 bits, only two of the 16 bits are actually used
- The LUT4 are populated with a default bit pattern, but the pattern is replaced, patched in a separate stage of the generation of the FPGA bitstream. This allows us to have device unique UDIs without generating a new FPGA bitstream for each device.